VSDiagnostics/VSDiagnostics/VSDiagnostics/Utilities/Extensions.cs

```diff
@@ -144,5 +145,37 @@ public static bool IsAsync(this IMethodSymbol methodSymbol)
                 || methodSymbol.ReturnType.MetadataName == typeof(Task).Name
                 || methodSymbol.ReturnType.MetadataName == typeof(Task<>).Name;
         }
+
+        public static bool IsDefinedInAncestor(this IMethodSymbol methodSymbol)
+        {
+            var containingType = methodSymbol?.ContainingType;
+            if (containingType == null)
+            {
+                return false;
+            }
+
+            var interfaces = containingType.AllInterfaces;
+            foreach (var @interface in interfaces)
+            {
+                var interfaceMethods = @interface.GetMembers().Select(containingType.FindImplementationForInterfaceMember);
+                if (interfaceMethods.Any(method => method.Equals(methodSymbol)))
+                {
+                    return true;
+                }
+            }
+
+            var baseType = containingType.BaseType;
+            while (baseType != null)
```

Comment:
I'm.. really surprised. We had that as an issue at some point (https://github.com/Vannevelj/VSDiagnostics/blob/develop/VSDiagnostics/VSDiagnostics/VSDiagnostics/Utilities/Extensions.cs#L63) and it had to be fixed by explicitly ending the loop if we're in `object`. More on this in the associated issue: https://github.com/Vannevelj/VSDiagnostics/issues/205

This is something we'll have to look into more closely but that can be done separately. I will create a separate issue for this, no need to add any tests. I will merge this PR since all thoughts I had have been cleared up, thanks for staying with me!